engines/python/setup/djl_python/huggingface.py

```diff
@@ -546,39 +575,45 @@ def update_adapter(inputs: Input):
     """
     Updates lora adapter with the model.
     """
-    if not is_rolling_batch_enabled(_service.hf_configs.rolling_batch):
-        raise NotImplementedError(
-            "LoRA adapter API is only supported for rolling batch.")
-
     adapter_name = inputs.get_property("name")
     adapter_alias = inputs.get_property("alias") or adapter_name
     adapter_path = inputs.get_property("src")
+    adapter_load = inputs.get_as_string(
+        "load").lower() == "true" if inputs.contains_key("load") else True
     adapter_pin = inputs.get_as_string(
         "pin").lower() == "true" if inputs.contains_key("pin") else False
 
     if adapter_name not in _service.adapter_registry:
         raise ValueError(f"Adapter {adapter_alias} not registered.")
 
     try:
+        if not adapter_load and adapter_pin:
+            raise ValueError("Can not set load to false and pin to true")
+
         old_adapter = _service.adapter_registry[adapter_name]
-        if old_adapter.get_property("src") and old_adapter.get_property(
-                "src") != adapter_path:
+        old_adapter_path = old_adapter.get_property("src")
+        if adapter_path != old_adapter_path:
             raise NotImplementedError(
                 f"Updating adapter path is not supported.")
 
-        old_adapter_pin = inputs.get_as_string(
-            "pin").lower() == "true" if inputs.contains_key("pin") else False
-        if old_adapter_pin and not adapter_pin:
-            raise NotImplementedError(f"Unpin adapter is not supported.")
-
-        if adapter_pin:
-            pinned = _service.rolling_batch.pin_lora(adapter_name)
-            if not pinned:
-                raise RuntimeError(
-                    f"Failed to pin LoRA adapter {adapter_alias}")
+        old_adapter_load = old_adapter.get_as_string("load").lower(
+        ) == "true" if old_adapter.contains_key("load") else True
+        if adapter_load != old_adapter_load:
+            if adapter_load:
+                _service.add_lora(adapter_name, adapter_alias, adapter_path)
+            else:
+                _service.remove_lora(adapter_name, adapter_alias)
```

Comment:
if adapter_load is false, why are we removing the adapter? Should this just be a noop?

Reply:
This is to have an unload option.

Reply:
I see, so we support unloading only for unpinned adapters

Reply:
Yes